com.arc.embeddedcdt/src/com/arc/embeddedcdt/common/DebuggerPluginVersionsChecker.java

```diff
@@ -0,0 +1,61 @@
+package com.arc.embeddedcdt.common;
+
+import java.util.HashSet;
+
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+
+import com.arc.embeddedcdt.LaunchConfigurationConstants;
+import com.arc.embeddedcdt.dsf.utils.Configuration;
+
+public class DebuggerPluginVersionsChecker {
+
+    private HashSet<String> seenLaunchConfigurationNames = new HashSet<>();
+
+    private static DebuggerPluginVersionsChecker INSTANCE = null;
+
+    private ILaunchConfiguration launchConfiguration;
+
+    private DebuggerPluginVersionsChecker() {
+    }
+
+    public static DebuggerPluginVersionsChecker getDebuggerPluginVersionsChecker() {
+        if (INSTANCE == null) {
+            INSTANCE = new DebuggerPluginVersionsChecker();
+        }
+        return INSTANCE;
+    }
+
+    public void checkPluginVersion(ILaunchConfiguration launchConfiguration) {
+        String launchConfigurationName = Configuration.getProgramName(launchConfiguration);
+        if (!seenLaunchConfigurationNames.contains(launchConfigurationName)) {
+            seenLaunchConfigurationNames.add(launchConfigurationName);
+            final int currentLaunchConfigurationDebuggerPluginVersion = Configuration
```

Comment:
Also local variable name is too long. It might make sense to have self-explanatory names for API entities, like functions or classes, but local variables can be shorter. `cfgFileFormatVersion` or something like that looks clear enough to mein the context. not sure if it should be  `cfgFileFormatVersion` or  `cfgFileFormatNumber` - up to you.